MapboxCoreNavigation/RouteController.swift

```diff
@@ -807,8 +825,12 @@ extension RouteController: CLLocationManagerDelegate {
         }
     }
 
-    func incrementRouteProgress() {
-        routeProgress.currentLegProgress.stepIndex += 1
+    func advanceStepIndex(to: Int? = nil) {
```

Comment:
Ideally, this method would take an `Array<RouteStep>.Index`, which would force callers to be safer and avoid potential out of bounds errors. However, that can be a bit of a pain to implement, so your call.